fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/FcrepoClientFactory.java

```diff
@@ -24,11 +24,24 @@
  */
 public class FcrepoClientFactory {
 	
-	private static String baseUri;
+	private String baseUri;
 	
-	public FcrepoClientFactory(String base) {
+	private static FcrepoClientFactory instance;
+	
+	private FcrepoClientFactory(String base) {
 		baseUri = base;
 	}
+	
+	public static FcrepoClientFactory factory(String baseUri) {
+		if (instance == null) {
+			instance = new FcrepoClientFactory(baseUri);
+		}
+		return instance;
+	}
+
+	public static FcrepoClientFactory defaultFactory() {
```

Comment:
What's the purpose of the defaultFactory() method? How does the instance's baseUri get set? Could the method return null?